platform/pulpcore/app/urls.py

```diff
@@ -1,73 +1,92 @@
 """pulp URL Configuration"""
-
-import warnings
-
-from django.core.exceptions import AppRegistryNotReady
 from django.conf.urls import url, include
 from rest_framework_nested import routers
 
 from pulpcore.app.apps import pulp_plugin_configs
 from pulpcore.app.views import ContentView, StatusView
 
-root_router = routers.DefaultRouter(
-    schema_title='Pulp API',
-    schema_url='/api/v3'
-)  #: The Pulp Platform v3 API router, which can be used to manually register ViewSets with the API.
-
-# Load non-nested viewsets first, so we can make nested routers with them.
-for app_config in pulp_plugin_configs():
-    for viewset in app_config.named_viewsets.values():
-        if not viewset.nest_prefix:
-            viewset.register_with(root_router)
-
-
-nested_routers = (
-    routers.NestedDefaultRouter(root_router, 'repositories', lookup='repository'),
-)
+import logging
+log = logging.getLogger(__name__)
 
 
-def router_for_nested_viewset(viewset):
+class VSNode(object):
     """
-    Find the router for a nested viewset.
-
-    Args:
-        viewset (pulpcore.app.viewsets.NamedModelViewSet): a viewset for which a nested router
-            should exist
-
-    Returns:
-        routers.NestedDefaultRouter: the nested router whose parent_prefix corresponds to the
-            viewset's nest_prefix attribute.
+    This is a tree that can register nested ViewSets with DRF routers.
```

Comment:
You describe this as a tree (and seems to behave like a tree).  Should this be named: `VSTree`?

Reply:
👍

Reply:
@jortel and I discussed this and decided on ViewSetNode